packages/ra-core/src/controller/usePrevNextController.ts

```diff
@@ -124,37 +125,78 @@ export const usePrevNextController = <RecordType extends RaRecord = any>(
         );
     }
 
-    const [storedParams] = useStore<StoredParams>(
+    const [storedParams] = useStore<Partial<ListParams>>(
         storeKey || `${resource}.listParams`,
         {
-            filter: {},
-            order: sort.order,
-            sort: sort.field,
+            filter: filterDefaultValues,
+            order: initialSort.order,
+            sort: initialSort.field,
+            page: 1,
+            perPage: 10,
         }
     );
 
-    const { data, error, isLoading } = useGetList<RecordType>(
+    const dataProvider = useDataProvider();
+    const queryClient = useQueryClient();
+    const pagination = { page: 1, perPage: limit };
+    const sort = {
+        field: storedParams.sort,
+        order: storedParams.order,
+    };
+    const filter = { ...storedParams.filter, ...permanentFilter };
```

Comment:
Shouldn't we rather use lodash merge here, in case the filters are defined on nested fields?

Reply:
Maybe, but since we do a simple merge as well in `useListController`, this would lead to different results than in the list, so we'd better keep it like that IMO. 

https://github.com/marmelab/react-admin/blob/923b63d85ad4deb20ad07aee440fe9160d63baf8/packages/ra-core/src/controller/list/useListController.ts#L93